datafusion/execution/src/memory_pool/memory_report.rs

```diff
@@ -0,0 +1,118 @@
+use super::{human_readable_size, MemoryReservation};
+use crate::memory_pool::pool::{
+    FairSpillPool, GreedyMemoryPool, TrackConsumersPool, UnboundedMemoryPool,
+};
+use crate::memory_pool::MemoryPool;
+use datafusion_common::Result;
+use datafusion_expr::Accumulator;
+use std::any::Any;
+
+/// Helper trait to provide memory usage breakdowns for debugging.
+///
+/// Implemented for [`MemoryReservation`] and any [`Accumulator`] via a blanket
+/// implementation that relies on [`Accumulator::size`].
+///
+/// # Example
+/// ```
+/// # use std::sync::Arc;
+/// # use datafusion_execution::memory_pool::{ExplainMemory, GreedyMemoryPool, MemoryConsumer};
+/// let pool = Arc::new(GreedyMemoryPool::new(1024));
+/// let mut reservation = MemoryConsumer::new("example").register(&pool);
+/// reservation.try_grow(256).unwrap();
+/// println!("{}", reservation.explain_memory().unwrap());
+/// ```
+pub trait ExplainMemory {
```

Comment:
I think it should include size as well so any trait that need to report the size will also can report the breakdown of the size.